### PR TITLE
fix(asm): fix another body read blocking problem on Flask

### PR DESCRIPTION
**Pull request by juanjux**
Signed-off-by: Juanjo Alvarez <juanjo.alvarezmartinez@datadoghq.com>

## Description
Fixed another potencial block on wsgi.input.read() on Flask on POSTed data.

In some cases, a read without a length can block, see:

https://stackoverflow.com/questions/22894078/why-does-environwsgi-input-read-block-even-though-it-is-allowed-by-pep-333

## Checklist
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Review by gnufede — APPROVED** [review]

**Review by avara1986 — APPROVED** [review]

**Review by christophe-papazian — DISMISSED** [review]

**Review by gnufede — DISMISSED** [review]

**Review by brettlangdon — CHANGES_REQUESTED** [review]
What is the actual problem? `wsgi_input` may be `None`?

Is there a regression test you can add?

The release note also isn't very descriptive to the issue being fixed.

It would be better as something like:

> ASM: do not try to read Flask ``wsgi_input`` if it is not available

**Comment by juanjux** [comment]
> What is the actual problem? `wsgi_input` may be `None`?
> 
> Is there a regression test you can add?
> 
> The release note also isn't very descriptive to the issue being fixed.
> 
> It would be better as something like:
> 
> > ASM: do not try to read Flask `wsgi_input` if it is not available

In some cases, a read without a length can block, see:

https://stackoverflow.com/questions/22894078/why-does-environwsgi-input-read-block-even-though-it-is-allowed-by-pep-333

Didn’t add a test because I wasn’t able to reproduce it locally but could logging on the reporter machine (also tested that the fix worked for him).

File: releasenotes/notes/fix-another-flask-body-blocking-acd533aa3c1bd46e.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    ASM: fix another body read blocking problem with Flask.
```
Comment (brettlangdon) [inline releasenotes/notes/fix-another-flask-body-blocking-acd533aa3c1bd46e.yaml:4]:
```suggestion
    ASM: do not block when reading from an empty ``wsgi_input`` with Flask.
```

not entirely sure if this is great, but I think it provides more context for anyone reading the release notes.

Comment (juanjux) [reply releasenotes/notes/fix-another-flask-body-blocking-acd533aa3c1bd46e.yaml]: Updated the changelog with another message, thanks for your feedback!

**Review by brettlangdon — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Review by avara1986 — APPROVED** [review]

**Comment by juanjux** [comment]
@mergifyio backport 1.5

**Comment by juanjux** [comment]
@mergifyio backport 1.6

**Pull request merged**